src/crags/resolvers/comments.resolver.ts

```diff
@@ -85,8 +87,19 @@ export class CommentsResolver {
 
   @AllowAny()
   @UseGuards(UserAuthGuard)
-  @Query(returns => [Comment], { name: 'exposedWarnings' })
+  @Query((returns) => [Comment], { name: 'exposedWarnings' })
   getExposedWarnings(@CurrentUser() user: User) {
     return this.commentsService.getExposedWarnings(user != null);
   }
+
+  @AllowAny()
+  @UseGuards(UserAuthGuard)
+  @Query((returns) => PaginatedComments, { name: 'latestComments' })
+  getLatestComments(
+    @Args('input', { type: () => FindCommentsInput })
+    input: FindCommentsInput,
+    @CurrentUser() user: User,
+  ) {
+    return this.commentsService.find(input, Boolean(user));
+  }
```

Comment:
I applied these two changes and the comments on the unpublished crags/routes are now filtered out (even to the publisher to keep it simple).

Reply:
I also decided to make a separate method in the comments service because I was not happy with applying all sorts of filtering, pagination and sorting logic to the `.find()` method of the `CommentsService`. The `.find()` method is used in some places where the unpublished comments should still be returned. However I decided to keep the [lines for filtering out comments on hidden crags](https://github.com/plezanje-net/api/pull/147/files#diff-adcf35b1d57a7194c4b3c11f00f5c0b00b5e27cdee932959ef3f31fa983e3a77R105-R113) in because it might still be useful.